### PR TITLE
net: Decouple DHCP server from converter pkg

### PR DESCRIPTION
**Pull request by orelmisan**
### What this PR does
The GetResolvConfDetailsFromPod function reads and parses `/etc/resolv.conf` - the DNS resolver's configuration file.

Move it from `pkg/virt-launcher/virtwrap/converter` to the `pkg/network/dns` package - under SIG network's ownership.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [X] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

File: pkg/network/dns/resolveconf.go [hunk 1]
```diff
@@ -122,3 +126,29 @@ func DomainNameWithSubdomain(searchDomains []string, subdomain string) string {
 
 	return ""
 }
+
+// returns nameservers [][]byte, searchdomains []string, error
```
Comment (EdDev) [inline pkg/network/dns/resolveconf.go:130]: Now that this is moved to a maintained location, please fix the function doc text.

File: pkg/network/dns/resolveconf.go [hunk 2]
```diff
@@ -122,3 +126,29 @@ func DomainNameWithSubdomain(searchDomains []string, subdomain string) string {
...
+		return nil, nil, err
+	}
+
+	searchDomains, err := ParseSearchDomains(string(b))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	log.Log.Reason(err).Infof("Found nameservers in %s: %s", resolvConf, bytes.Join(nameservers, []byte{' '}))
+	log.Log.Reason(err).Infof("Found search domains in %s: %s", resolvConf, strings.Join(searchDomains, " "))
```
Comment (EdDev) [inline pkg/network/dns/resolveconf.go:151]: I do not like these logs, but we can think about this in the future.

**Review by EdDev — APPROVED** [review]
Thanks you!

Approving the network part:
/approve

File: pkg/network/dns/resolveconf.go (hunk 1 above)
Comment (orelmisan) [reply pkg/network/dns/resolveconf.go]: Done.

**Review by enp0s3 — COMMENTED** [review]
@orelmisan Thank you!
/approve

**Pull request merged**